### ISSUE TITLE
BUG ResetInputAjaxActionListener with Mojarra 3.0.2 ajax action param is null

### ISSUE THREAD

**Issue by pizzi80** [omnifaces/omnifaces#683]
I'm having really bad time to debug this, at the moment I can say that:

if I activate the `ResetInputAjaxActionListener` inside `faces-config.xml`

```
<application>
  <action-listener>org.omnifaces.eventlistener.ResetInputAjaxActionListener</action-listener>
</application>
```

the following example do not works on Mojarra 3.0.2 with OmniFaces 4.0-M13 :

with Mojarra 3.0.0 works
with Mojarra 3.0.1 works
with Mojarra 3.0.2 **fail**   **with NPE because pojo is null !!**

```
<h:panelGroup id="data" layout="block" styleClass="card" >
    <h:form id="testInsideList">

        <ui:repeat id="list" value="#{test.list}" var="_pojo">
            <h:panelGroup id="pojo" layout="block" >

                #{_pojo}

                <h:commandButton action="#{test.action(_pojo)}" value="Action and render ajax" >
                    <f:ajax execute="@this" render=":data" />
                </h:commandButton>

            </h:panelGroup>
        </ui:repeat>
    </h:form>
</h:panelGroup>
```

```
@Named
@ViewScoped
public class Test implements Serializable {

  @Serial private static final long serialVersionUID = 8156941558380042283L;

  private static final Logger log = Logger.getLogger(Test.class.getName());
  
  private List<Pojo> list;

  public List<Pojo> getList() {return list;}
  
  @PostConstruct
  public void init() {
    list = new ArrayList<>();
    list.add( new Pojo(1,"Foo") );
    list.add( new Pojo(2,"Bar") );
  }

  public void action(Pojo pojo) {
    
    pojo.setName(pojo.getName()+" clicked");
    
    log.info("action > "+pojo );
  }
}
```
```
public class Pojo implements Serializable {

  @Serial private static final long serialVersionUID = -582308979803055368L;

  private int id;  
  private String name;
     
  
  public Pojo() {
    	  super();
  }
  
  public Pojo(int id, String name) {
	  this.id = id;
	  this.name = name;
  }
  
  @Override
  public boolean equals(Object o) {
	  if (this == o) return true;
	  if (o == null || getClass().isAssignableFrom(o.getClass()) ) return false;
	  Pojo pojo = (Pojo) o;
	  return id == pojo.id;
  }

  @Override public int hashCode() {return Objects.hash(id);}
  @Override public String toString() {return "Pojo{id=" + id + ", name='" + name + '\'' + '}';}

  public int getId() {return id;}
  public void setId(int id) {this.id = id;}
  public String getName() {return name;}
  public void setName(String name) {this.name = name;}
}


```

**Comment by BalusC** [comment]
Reproduced. But happens for me also in 3.0.0 / 3.0.1. Then I tried 2.3.x and the 2.3.15 was the first version exposing this issue. It works in 2.3.14 and before.

The tree visit performed by ResetInputAjaxActionListener has unintentionally cleared out some state of the ui:repeat. This is most probably related to [this change](https://github.com/eclipse-ee4j/mojarra/commit/6e67eaf8cbdf3947a9da575561db844f078fce68) in 2.3.15: https://github.com/eclipse-ee4j/mojarra/issues/4830
